### DISCUSSION TITLE
Armature molecules

### DISCUSSION THREAD

**Discussion by tschiex** [BradyAJohnston/MolecularNodes#126] [Ideas]
**Is your feature request related to a problem? Please describe.**

MolecularNodes relies on Cartesian coordinates but it's often far more comfortable to use internal coordinates (using angles and distances). In Blender, armatures essentially offer this facility. It would be nice if a protein could be equipped with a proper armature such that one could control the backbone (phi/psi/omega) and the side chains (ch1 to chi4, depe,nding on the amino acid)

This would also be handy for interpolation between RMN or MD states. Instead of interpolating Cartesian coordinates, the interpolation of angles would essentially avoid the nasty intermediates states you show in your YouTube video.

**Comment by BradyAJohnston** [comment]
This is an interesting point for using internal coordinates. 

I have avoided doing anything with armatures so far because armatures and rigging is something that is on the roadmap for being overhauled in Blender and being properly supported inside of Geometry Nodes. Once the system is set up more comprehensively inside of Geometry Nodes then it would be excellent to have that kind of functionality.

The `Wiggle Amino Acids` which I show off [here](https://twitter.com/bradyajohnston/status/1566792994107191296?s=20&t=2u68HY6MidhWnTq4h7mmlQ) uses this kind of approach. I progressively move out along the side chains and rotate around particular angles. It would certainly be improved by the addition of proper armature support though.

There is potential to add support for armatures outside of Geometry Nodes, using the bone system that is currently inside of Blender. I haven't put time towards it because it will eventually be replaced, but if somebody wanted to have a go at implementing something with it I would be more than happy to accept PR on the topic!

**Comment by tschiex** [comment]
Thanks for the feedback Brady. I will definitely have a look at these Wiggling amino acids (which I missed yesterday, although I spent all afternoon following your videos :-) ), I will definitely buy you a coffee ASAP for all these amazing and very elegant efforts (previously supported BioBlender, but this is far more elegant, thanks to Geometry Nodes).
I will be waiting for the progress of rigging inside geometry nodes for sure! Keep up the good job.

**Comment by BradyAJohnston** [comment]
Thanks for the kind words and the coffee! BioBlender was headed in the right direction but I'm glad that I was able to figure out a more integrated method for getting it all working.

I'm having regular discussions with the Blender devs about what would be best for molecular data inside Geometry Nodes. They love what we are doing with it ([I gave a talk at BCON this year](https://youtu.be/adhTmwYwOiA)) and they are excited to try and support us in doing more protein vis as much as they can. Hopefully this means that more streamlined rotations of amino acids and protein backbones are on the way!
